se_shared/str_utils.h

```diff
@@ -0,0 +1,46 @@
+#ifndef STR_UTILS_H
+#define STR_UTILS_H
+
+#include <sstream>
+#include <string>
+#include <vector>
+
+/**
+ * Split a string into a vector of substrings based on a delimiter.
+ *
+ * @note The contents of `elems` are not deleted, substrings are just pushed
+ * back.
+ *
+ * \param[in] s The string to split.
+ * \param[in] delim The delimiter to use.
+ * \param[out] elems The vector that will contain the substrings.
+ * \return A reference to the vector containing the substrings.
+ */
+static std::vector<std::string>& splitString(const std::string&        s,
```

Comment:
I am not sure I see the reason for this. Is it used anywhere?

Reply:
It was implemented like this in an older version of supereight. Kept only the function returning by value in d0e0cf4dcde9e2be2904002f599c9a285638de8c.